aws/rust-runtime/aws-sigv4/Cargo.toml

```diff
@@ -23,10 +23,17 @@ http = { version = "0.2", optional = true }
 once_cell = "1.8"
 percent-encoding = { version = "2.1", optional = true }
 regex = "1.5"
-ring = "0.16"
 time = "0.3.5"
 tracing = "0.1"
 
+[target.'cfg(not(any(target_arch = "powerpc", target_arch = "powerpc64")))'.dependencies]
+ring = "0.16"
+
+# ring does not compile on powerpc (https://github.com/briansmith/ring/issues/389) so we use hmac and sha2.
+[target.'cfg(any(target_arch = "powerpc", target_arch = "powerpc64"))'.dependencies]
+hmac = "0.12"
+sha2 = "0.10"
```

Comment:
Considering that the implementation using `hmac` and `sha2` can run on all platforms, do you think it makes sense for us to keep `ring` around and juggle the burden of conditional compilation here?

Reply:
AFAIK ring is much more performant on x86 than hmac + sha2. I'll make some benches to ensure we select the right dependency.

Reply:
If we keep Ring for non-powerpc, you might want to add the reason in the comment here.

Reply:
I have added some benchmarks and ring is 3x slower than hmac+sha2. This PR is still in the works. I am talking with Jon to understand if we want to migrate all the architectures away from ring.